offline/cable_cru_TRENDY.F90

```diff
@@ -29,22 +29,41 @@ MODULE CABLE_CRU
 
 IMPLICIT NONE
 
+! Set some private parameters
+! To start the work on generalising the Met and subdiurnalisation process,
+! we give each possible variable an index to refer to in the subdiurnalisation
+! routine. The namelist can accept files for every variable, with unsupplied
+! filenames defaulting to empty strings. Since we're still passing every
+! possible file name to the Dataset generator, we need a consistent string
+! to stop the building process.
+REAL, PRIVATE, PARAMETER  :: SecDay = 86400.
+INTEGER, PRIVATE, PARAMETER  :: rain = 1, lwdn = 2, swdn = 3, pres = 4,&
+                                qair = 5, tmax = 6, tmin = 7, uwind = 8,&
+                                vwind = 9, wind = 10, vph0900 = 11, vph1500 = 12,&
+                                fdiff = 13, prevTmax = 14, nextTmin = 15,&
+                                nextvph0900 = 16, prevvph1500 = 17
+INTEGER, PRIVATE, PARAMETER  :: sp = kind(1.0)
+INTEGER, PRIVATE, PARAMETER  :: dp = KIND(1.0d0)
+INTEGER, PRIVATE, PARAMETER  :: nVariables = 13
```

Comment:
I think the best way is to add some commentary. The nVariables is the number of "readers" we need, since we don't need to open another reader for the next day/previous day variables. The difference between the numbers of variables we actually need vs the number of datasets we need open is the reason for the ```nExtraVariables``` in the next line. Most of the iteration loops are based on this ```nVariables```, but we need storage for ```nVariables + nExtraVariables``` arrays of variables. Perhaps a rename of the variables? ```nCoreVariables``` and ```nDerivedVariables```? But even then, they're not "derived" as such.